docs/table-upgrade.md

```diff
@@ -0,0 +1,41 @@
+# Table Upgrade logic and data structures
+The Hive Metastore migration process will upgrade the following Assets:
+- Tables ond DBFS root
+- External Tables
+- Views
```

Comment:
what about other entities, like functions? do we want to migrate those?

Reply:
If a view uses a function, we need to expose whether or not that function CAN be migrated.

Reply:
To be super clear, the function has to be migratable and we have to make that assessment as part of this, otherwise, the view that uses it can't be migrated.

Reply:
We will make a "best effort" to begin with we should mark these.